Assignment 3.Rmd

```diff
@@ -127,29 +119,115 @@ Now we can plot our line plot using the ggplot command, "ggplot()".
 
-##Part II
+-- Clustering by 3 is more informative, because people are grouped in 3 groups that have high, medium, and low motivation.
+```{r}
+fitnew <- kmeans(K3,3)
+fitnew$cluster
+K4new <- data.frame(K3, fitnew$cluster)
+names(K4new) <- c("1","2","3","4","5","cluster")
+K5new <- tidyr::gather(K4new,"week","motivation",1:5)
+K6new <- K5new%>%dplyr::group_by(cluster,week)%>%summarise(mean(motivation))
+K6new$week <- as.numeric(K6new$week)
+K6new$cluster <- as.factor(K6new$cluster)
+ggplot(K6new,aes(x = week, y = K6new$`mean(motivation)`, color = cluster)) + geom_line()+ xlab("Week")+ylab("Average Motivation")
+K7new <- K4new%>%count(cluster)
+```
+
+
 
+##Part II
 Using the data collected for Assignment 2 (which classes students were in), cluster the students, then redraw the graph of the class but color the students according the cluster they are in.
+```{r}
+library(dplyr)
+library(tidyr)
+h1 <- read_csv("hudk4050-classes.csv")
+h1 <- h1%>% unite(Name, `First Name`, `Last Name`, sep= " ")
+h2 <- gather(h1,key="class number", value="class",2:7)
+h2 <- na.omit(h2)
+h2 = select(h2, -c("class number"))
+h2$class <- gsub("[[:space:]]", "", h2$class)
+h2 <- h2%>%filter(Name!='ZIMO CHEN')
+h2 <- h2%>%filter(class!="HUDK4050")
+h2$count=1
+h3 <- spread(h2,class,count)
+h3[is.na(h3)] <- 0
+rownames(h3) <- h3$Name
+h4 <- h3[,2:59]
+h4 <- t(t(h4)) %*% t(h4)
+colnames(h4) <- h3$Name
+rownames(h4) <- h3$Name
+#h3 is the person-class matrix, h4 is the person-person matrix
+h3_noname<- h3[,-1]
+h5 <- scale(h3_noname)
+fit2 <- kmeans(h5,3)
+fit2$cluster
+h5 <- data.frame(h5, fit2$cluster)
+h5$fit2.cluster <- as.factor(h5$fit2.cluster)
+
+library(igraph)
+g<-graph.adjacency(h4,mode="undirected")
+plot(g, layout=layout.fruchterman.reingold, vertex.color=h5$fit2.cluster)
+```
 
-##Part III
 
+##Part III
 In class activity 6 you clustered students in the class by the answers to a questionaire. Create a visualization that shows the overlap between these clusters and the clusters generated in part II.
 
+```{r}
+DF1 <- read_csv("HUDK405019-clustering.csv")
+DF1 <- DF1%>%unite(Name,`First Name`,`Last Name`,sep=" ") 
+rownames(DF1) <- DF1$Name
+#Wrangle data using dplyr to include only the numerical values.
+DF2<-DF1%>%mutate_all(as.character)
+DF3<-DF2%>%mutate_all(as.numeric)
+DF4<-DF3[,-1]
+DF4<-DF4[,-12]
+DF4<-DF4[,-12:-13]
+rownames(DF4) <- DF1$Name
+DF4[is.na(DF4)] <- 0
+#Scale the data so that no variable has undue influence
+DF4 <- scale(DF4)
+
+fit3 <- kmeans(DF4,3)
+DF4 <- data.frame(DF4, fit3$cluster)
+DF5 <- as.data.frame(cbind(toupper(row.names(DF4)),DF4$fit3.cluster))
+colnames(DF5) <- c("name","questionairCluster")
+
+rownames(h5) <- h3$Name
+h6 <-as.data.frame(cbind(toupper(row.names(h5)),h5$fit2.cluster))
+colnames(h6) <- c("name","classCluster")
+
+G1 <- merge(DF5,h6,by="name")
+
+## overlap
+# install.packages("vcd")
+library(grid)
+library(vcd)
+S1<-structable(G1$questionairCluster~G1$classCluster)
+mosaic(S1,shade=TRUE,legend=TRUE)
+
```

Comment:
Well done on using the mosaic plot. Overall, great job on the coding aspect of the assignment but you may need to to work on the analysis part. Nonetheless, keep up the good work!